scripts/apps/archive/directives/MarkedItemTitle.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * @module superdesk.apps.archive
+ * @ngdoc directive
+ * @name sdMarkedItemTitle
+ * @requires desks
+ * @requires authoring
+ * @requires highlightServce
+ * @requires $timeout
+ * @description
+ *   This directive is used in authoring-topbar to mark a story as a highlight or
+ *   to mark it for a desk.
+ */
+MarkedItemTitle.$inject = ['desks', '$timeout', 'authoring', 'highlightsService'];
+export function MarkedItemTitle(desks, $timeout, authoring, highlightsService) {
+    return {
+        scope: {
+            item: '=item',
+            markField: '@field'
+        },
+        templateUrl: 'scripts/apps/archive/views/marked_item_title.html',
+        // todo(petr): refactor to use popover-template once angular-bootstrap 0.13 is out
+        link: function(scope, elem) {
```

Comment:
This link function is way too long imo and is hard to follow the logic. Can we maybe create a class and split all this logic into multiple smaller (documented) functions?

For directives you need to `return new ClassName()` and it will work fine. See [this example] from web publisher.